File: code/native/kotlin/runtime/io.kt
import java.io.File

object io {
    fun fileRead(path: String): String = try { File(path).readText() } catch (e: Exception) { "" }
    fun fileWrite(path: String, data: String) { try { File(path).writeText(data) } catch (e: Exception) {} }
    fun fileAppend(path: String, data: String) { try { File(path).appendText(data) } catch (e: Exception) {} }
    fun fileRemove(path: String) { try { File(path).delete() } catch (e: Exception) {} }
    fun fileCopy(from: String, to: String) { try { File(from).copyTo(File(to), overwrite = true) } catch (e: Exception) {} }
    fun fileMove(from: String, to: String) { try { File(from).copyTo(File(to), overwrite = true); File(from).delete() } catch (e: Exception) {} }
    fun fileExists(path: String): Boolean = File(path).exists()
    fun fileSize(path: String): Long = try { File(path).length() } catch (e: Exception) { 0L }
    fun isDirectory(path: String): Boolean = File(path).isDirectory
    fun isFile(path: String): Boolean = File(path).isFile
    fun dirMake(path: String) { try { File(path).mkdirs() } catch (e: Exception) {} }
    fun dirRemove(path: String) { try { File(path).deleteRecursively() } catch (e: Exception) {} }
}

File: code/native/rust/runtime/io.rs
mod io {
    pub fn file_read(path: String) -> String {
        std::fs::read_to_string(&path).unwrap_or_default()
    }
    pub fn file_write(path: String, data: String) {
        let _ = std::fs::write(&path, data);
    }
    pub fn file_append(path: String, data: String) {
        use std::io::Write;
        if let Ok(mut f) = std::fs::OpenOptions::new().create(true).append(true).open(&path) {
            let _ = f.write_all(data.as_bytes());
        }
    }
    pub fn file_remove(path: String) {
        let _ = std::fs::remove_file(&path);
    }
    pub fn file_copy(from: String, to: String) {
        let _ = std::fs::copy(&from, &to);
    }
    pub fn file_move(from: String, to: String) {
        let _ = std::fs::rename(&from, &to);
    }
    pub fn file_exists(path: String) -> bool {
        std::path::Path::new(&path).exists()
    }
}
